registration/controllers/root.py

```diff
@@ -18,8 +18,10 @@ class RootController(TGController):
     @expose('genshi:registration.templates.register')
     @expose('kajiki:registration.templates.register')
     def index(self, *args, **kw):
+        hooks.notify('registration.before_index', args=(kw,))
```

Comment:
`before_index` is not much meaningful, maybe `before_registration_form` better conveys the purpose.
Also note that prooobably in this case you actually wanted to subclass the form and go through prepare. But I can see that this is easier.

In any case we probably don't want to set the argument into the action url, but we want to add an hidden field to the form so that it is retained also in case of validation errors etc...